File: 2020/test/day_05_test.exs
defmodule Day05Test do
  use ExUnit.Case

  import Day05
  import TestHelper

  describe "Day05" do
    test "part1" do
      assert part1(input("day_05_example_1")) == 820
      assert part1(input("day_05")) == 871
    end

    test "part2" do
      assert part2(input("day_05")) == 640
    end
  end
end

File: 2021/test/day_05_test.rb
# frozen_string_literal: true

require "test_helper"
require "day_05"

class Day05Test < Minitest::Test
  def test_part1
    day = Day05.new

    assert_equal 5, day.part1(input("day_05_example_1"))
    assert_equal 5774, day.part1(input("day_05"))
  end

  def test_part2
    day = Day05.new

    assert_equal 12, day.part2(input("day_05_example_1"))
    assert_equal 18_423, day.part2(input("day_05"))
  end
end
